### ISSUE TITLE
Local clusterfuzz failed to install dependencies caused by version of GLIBC on Ubuntu 18.04

### ISSUE THREAD

**Issue by leonasdev** [google/clusterfuzz#2835]
## Faulty behavior
When installing dependencies `local/install_deps.bash` following [Prerequisites](https://google.github.io/clusterfuzz/getting-started/prerequisites/), the command `npm install --unsafe-perm -g bower polymer-bundler` result in an error.
Screenshot:
![image](https://user-images.githubusercontent.com/39915562/198813224-f554b3eb-ae24-40a9-a702-bcce8b41f1f0.png)

## Error
```
node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by node)
```

## Related
#2644 

## Reproduction
- Host: Ubuntu 18.04 on WSL 2.0
- Clusterfuzz: `git clone https://github.com/google/clusterfuzz && git checkout tags/v2.5.7`
- Python: 3.7.7 from source from official repo with following step:
  ```
  RUN apt-get install -y apt-transport-https software-properties-common build-essential git curl \
      libssl-dev zlib1g-dev libncurses5-dev libncursesw5-dev libreadline-dev libsqlite3-dev \
      libgdbm-dev libdb5.3-dev libbz2-dev libexpat1-dev liblzma-dev libffi-dev uuid-dev
  RUN curl -sS https://www.python.org/ftp/python/3.7.7/Python-3.7.7.tgz | tar -C /tmp -xzv && \
      cd /tmp/Python-3.7.7 && \
      ./configure --enable-optimizations && sudo make altinstall && \
      sudo rm -rf /tmp/Python-3.7.7
  ```
- Go: v1.19.2, install as described in [docs](https://go.dev/doc/install)

Next we run `local/install_deps.bash`. This runs result in an error as shown below:
### Version 'GLIBC 2.28' Not Found
```
node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by node)
```

## Expected behavior
All dependencies should be correctly installed with no errors.

## Scope
The host OS Ubuntu 18.04 which is described as a supported system in [docs](https://google.github.io/clusterfuzz/getting-started/prerequisites/#other-dependencies). But with a default Ubuntu 18.04, all dependencies cannot be installed correctly.

**Comment by oliverchang** [comment]
Thanks for filing! Would it be feasible to upgrade the distro? We're not able to feasibly support older versions, and we need to update our docs.

**Comment by leonasdev** [comment]
> Thanks for filing! Would it be feasible to upgrade the distro? We're not able to feasibly support older versions, and we need to update our docs.

@oliverchang If you run all dependencies command except `npm install --unsafe-perm -g bower polymer-bundler`, then upgrade the distro, it will work.

But if you run `local/install_deps.bash` in a brand new Ubuntu 22.04 distro, you will get more errors.

**Comment by leonasdev** [comment]
> Thanks for filing! Would it be feasible to upgrade the distro? We're not able to feasibly support older versions, and we need to update our docs.

@oliverchang After I upgrade the distro to Ubuntu 22.04, and re-run `local/install_deps.bash`, the previous issue is solved, but it comes to a new issue:
```
Running: pip3 install -r src/requirements.txt --upgrade --target src/third_party| Looking in indexes: https://pypi.python.org/simple
| Collecting antlr4-python3-runtime==4.7.2
|   Using cached antlr4-python3-runtime-4.7.2.tar.gz (112 kB)
|   Preparing metadata (setup.py): started
|   Preparing metadata (setup.py): finished with status 'error'
|   error: subprocess-exited-with-error
|
|   × python setup.py egg_info did not run successfully.
|   │ exit code: 1
|   ╰─> [1 lines of output]
|       ERROR: Can not execute `setup.py` since setuptools is not available in the build environment.
|       [end of output]
|
|   note: This error originates from a subprocess, and is likely not a problem with pip.
| error: metadata-generation-failed
|
| × Encountered error while generating package metadata.
| ╰─> See above for output.
|
| note: This is an issue with the package mentioned above, not pip.
| hint: See above for details.
| WARNING: You are using pip version 22.0.3; however, version 22.3 is available.| You should consider upgrading via the '/home/selab-fuzz/.local/share/virtualenvs/clusterfuzz-e6SeqTvW/bin/python -m pip install --upgrade pip' command.
| Return code is non-zero (1).
| Exit.
```
Do you know how to solve it?

**Comment by Alan32Liu** [comment]
Hi @leonasdev, I was looking into it but could not reproduce it with a clean `Ubuntu20:04` and `Python3.8`.
It appears that the error is about installing `antlr4-python3-runtime` in your `Python` virtual environment, could you please check if there is any issue with the env? (e.g. is it clean, etc.).

**Comment by leonasdev** [comment]
> Hi @leonasdev, I was looking into it but could not reproduce it with a clean `Ubuntu20:04` and `Python3.8`. It appears that the error is about installing `antlr4-python3-runtime` in your `Python` virtual environment, could you please check if there is any issue with the env? (e.g. is it clean, etc.).

In `Ubuntu 20.04`, all commands in install_deps.bash executed successfully but with one problem: `python3.8 butler.py run_server --bootstrap` will show an error with the JRE version, which can be solved by: `sudo apt install openjdk-11-jdk`.

But if you upgrade the distro from 18.04 to 22.04, you will encounter the problem I mention before.

I think we should update the docs and add a dependency of JRE in `install_deps_linux.bash` then we can close this issue.
@oliverchang @Alan32Liu

**Comment by Alan32Liu** [comment]
Thanks, @leonasdev!

> In Ubuntu 20.04, all commands in install_deps.bash executed successfully but with one problem: python3.8 butler.py run_server --bootstrap will show an error with the JRE version, which can be solved by: sudo apt install openjdk-11-jdk.

Yep, I can confirm that the error can be reproduced and fixed in a clean `Ubuntu:20.04`.
Let me add it to `install_deps.bash` as you suggested.

> But if you upgrade the distro from 18.04 to 22.04, you will encounter the problem I mention before.

Noted, I will pay attention to that when we upgrade to `22.04` in the future.
